cluster-autoscaler/FAQ.md

```diff
@@ -297,140 +308,143 @@ kubectl annotate node <nodename> cluster-autoscaler.kubernetes.io/scale-down-dis
+
+What happens when a non-empty node is deleted? As mentioned above, all pods should be migrated
+elsewhere. Cluster Autoscaler does this by evicting them and tainting the node, so they aren't
+scheduled there again.
+
+Example scenario:
+
+Nodes A, B, C, X, Y.
+A, B, C are below utilization threshold.
+In simulation, pods from A fit on X, pods from B fit on X, and pods from C fit
+on Y.
+
+Node A was deleted. OK, but what about B and C, which were also eligible for deletion? Well, it depends.
+
+Pods from B may no longer fit on X after pods from A were moved there. Cluster Autoscaler has to find place for them somewhere else, and it is not sure that if A had been deleted much earlier than B, there would always have been a place for them. So the condition of having been unneded unneeded for 10 min may not be true for B anymore.
+
+But for node C, it's still true as long as nothing happened to Y. So C can be deleted immediately after A, but B may not.
 
 Cluster Autoscaler does all of this accounting based on the simulations and memorized new pod location.
-They may not always be precise (pods can land elsewhere) but it seems to be a good heuristic so far.
+They may not always be precise (pods can be scheduled elsewhere in the end), but it seems to be a good heuristic so far.
 
-### Does CA work with PodDisruptionBudget in scale down?
+### Does CA work with PodDisruptionBudget in scale-down?
 
-From 0.5 CA (K8S 1.6) respects PDB. Before starting to delete a node CA makes sure that there is at least some non-zero PodDisruptionBudget. Then it deletes all pods from a node through the pod eviction api, retrying, if needed, for up to 2 min. During that time other CA activities are stopped. If one of the evictions fails the node is saved and it is not deleted, but another attempt to delete it may be conducted in the near future.
+From 0.5 CA (K8S 1.6) respects PDBs. Before starting to delete a node, CA makes sure that PodDisruptionBudgets for pods scheduled there allow for removing at least one replica. Then it deletes all pods from a node through the pod eviction API, retrying, if needed, for up to 2 min. During that time other CA activity is stopped. If one of the evictions fails, the node is saved and it is not deleted, but another attempt to delete it may be conducted in the near future.
 
-### Does CA respect GracefulTermination in scale down?
+### Does CA respect GracefulTermination in scale-down?
 
-CA, from version 1.0, gives pods at most 10 min graceful termination time. If the pod is not stopped within
-these 10 min then the node is deleted anyway. Earlier versions of CA gave 1 min or didn't respect graceful
-termination at all.
+CA, from version 1.0, gives pods at most 10 minutes graceful termination time. If the pod is not stopped within these 10 min then the node is deleted anyway. Earlier versions of CA gave 1 minute or didn't respect graceful termination at all.
 
 ### How does CA deal with unready nodes in version <= 0.4.0?
```

Comment:
Do we still need that question? 0.4.0 would be k8s 1.5 - what's our policy on support period for CA?

Reply:
Removed as discussed offline with @mwielgus